### PR TITLE
feat(expo): Added ability to provide sentryUrl for source maps upload

### PR DESCRIPTION
**Pull request by Acetyld**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [X] Enhancement
- [ ] Refactoring

## :scroll: Description
Added sentryUrl for people that are self hosting

## :bulb: Motivation and Context
Its also possible to define URL inside app.json and this was missing from the command.

## :green_heart: How did you test it?
I tested it locally.

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   },
   "typescript.tsdk": "./node_modules/typescript/lib",
   "editor.codeActionsOnSave": {
-    "source.fixAll": true
+    "source.fixAll": "explicit"
```
Comment (krystofwoldrich) [inline .vscode/settings.json:17]: This change is unrelated to the PR. Suggesting, revert.

```suggestion
    "source.fixAll": true
```

File: scripts/expo-upload-sourcemaps.js [hunk 2]
```diff
@@ -147,6 +149,13 @@ if (!sentryOrg || !sentryProject) {
     console.log(`${SENTRY_PROJECT} resolved to ${sentryProject} from expo config.`);
   }
 }
+if (pluginConfig.url) {
+  sentryUrl = pluginConfig.url;
+} else {
+  sentryUrl
+}
+console.log(`${SENTRY_PROJECT} resolved to ${sentryProject} from expo config.`);
+
```
Comment (krystofwoldrich) [inline scripts/expo-upload-sourcemaps.js:158]: Although, I agree with loading the URL from the Expo Plugin Configuration.

The current code won't work, as `pluginConfig` variable is not in the scope.

This should be under `if (!sentryOrg || !sentryProject) {`, which needs to be updated to include `sentryUrl`.

File: scripts/expo-upload-sourcemaps.js [hunk 3]
```diff
@@ -196,8 +206,7 @@ if (numAssetsUploaded === totalAssets) {
   console.log('✅ Uploaded bundles and sourcemaps to Sentry successfully.');
 } else {
   console.warn(
-    `⚠️  Uploaded ${numAssetsUploaded} of ${totalAssets} bundles and sourcemaps. ${
-      numAssetsUploaded === 0 ? 'Ensure you are running `expo export` with the `--dump-sourcemap` flag.' : ''
+    `⚠️  Uploaded ${numAssetsUploaded} of ${totalAssets} bundles and sourcemaps. ${numAssetsUploaded === 0 ? 'Ensure you are running `expo export` with the `--dump-sourcemap` flag.' : ''
```
Comment (krystofwoldrich) [inline scripts/expo-upload-sourcemaps.js:209]: Please avoid changes that are not related to the PR.

**Comment by krystofwoldrich** [comment]
@Acetyld Hi, thank you for the PR,

this makes sense and is a good addition to the Expo script.

But before we can add it the code needs a few fixes that I described in the comment above.

**Comment by Acetyld** [comment]
Hi =) Yhea i used the browser vscode editor, it was a bit wonky.
Sry for committing above unrelated changed ^^

I saw u fixed almost all issue's so i suppose it is fine now?

File: scripts/expo-upload-sourcemaps.js (hunk 3 above)
Comment (krystofwoldrich) [reply scripts/expo-upload-sourcemaps.js:209]:
```suggestion
    `⚠️  Uploaded ${numAssetsUploaded} of ${totalAssets} bundles and sourcemaps. ${
      numAssetsUploaded === 0 ? 'Ensure you are running `expo export` with the `--dump-sourcemap` flag.' : ''
```

Comment (krystofwoldrich) [inline scripts/expo-upload-sourcemaps.js:209]:
```suggestion
    `⚠️  Uploaded ${numAssetsUploaded} of ${totalAssets} bundles and sourcemaps. ${
      numAssetsUploaded === 0 ? 'Ensure you are running `expo export` with the `--dump-sourcemap` flag.' : ''
```

**Comment by krystofwoldrich** [comment]
@Acetyld There is just this one [comment to be resolved](https://github.com/getsentry/sentry-react-native/pull/3664#discussion_r1516123691).

**Comment by Acetyld** [comment]
Alright i tried to update above, if it still incorrect i will put time into doing this in a real editor instead of the git edit 😜

**Review by krystofwoldrich — APPROVED** [review]
@Acetyld Thank you, now it's perfect. 🚀
